editor.planx.uk/src/routes/authenticated.tsx

```diff
@@ -11,14 +14,45 @@ import { makeTitle } from "./utils";
 
 const editorRoutes = compose(
   withData(() => ({
-    // just putting anything here for now
-    username: "A",
+    username: useStore.getState().getUser().firstName,
   })),
 
   withView(() => <AuthenticatedLayout />),
 
   mount({
     "/": route(async () => {
+      const jwt = getCookie("jwt");
+      const email = jwt && (jwtDecode(jwt) as any)["email"];
+      const users = await client.query({
+        query: gql`
+          query GetUserByEmail($email: String!) {
+            users: users(where: { email: { _eq: $email } }) {
+              id
+              firstName: first_name
+              lastName: last_name
+              email
+              isPlatformAdmin: is_platform_admin
+              teams {
+                role
+                team {
+                  name
+                  slug
+                  id
+                }
+              }
+            }
+          }
+        `,
+        variables: { email },
+      });
```

Comment:
I'm using the Apollo client for authenticated users here out of convenience, but this query is also duplicated in planx-core's `UserClient` as `getByEmail` - likely one to simplify going forward!